maven-plugin/src/main/java/org/jboss/galleon/maven/plugin/util/AbstractMavenArtifactRepositoryManager.java

```diff
@@ -100,6 +98,43 @@ public void resolve(MavenArtifact artifact) throws MavenUniverseException {
         artifact.setPath(Paths.get(result.getArtifact().getFile().toURI()));
     }
 
+    @Override
+    public void resolveAll(List<MavenArtifact> artifacts) throws MavenUniverseException {
+        if (artifacts.stream().anyMatch(MavenArtifact::isResolved)) {
+            throw new MavenUniverseException("One of artifact is already resolved");
+        }
+
+        List<ArtifactRequest> requests = new ArrayList<>();
+        for (MavenArtifact artifact : artifacts) {
+            requests.add(toArtifactRequest(artifact));
+        }
+
+        try {
+            final List<ArtifactResult> artifactResults = repoSystem.resolveArtifacts(getSession(), requests);
+            for (int i = 0, artifactResultsSize = artifactResults.size(); i < artifactResultsSize; i++) {
```

Comment:
artifactResultsSize doesn't seem required, you could use artifactResults.size().